compiler/rustc_const_eval/src/util/compare_types.rs

```diff
@@ -24,16 +24,22 @@ pub fn is_equal_up_to_subtyping<'tcx>(
     }
 
     // Check for subtyping in either direction.
-    is_subtype(tcx, param_env, src, dest) || is_subtype(tcx, param_env, dest, src)
+    relate_types(tcx, param_env, Variance::Covariant, src, dest)
+        || relate_types(tcx, param_env, Variance::Covariant, dest, src)
 }
 
 /// Returns whether `src` is a subtype of `dest`, i.e. `src <: dest`.
 ///
+/// For almost all of the use cases variance should be `Covariant`,
+/// in `MirPhase` >= `MirPhase::Runtime(RuntimePhase::Initial` variance should
+/// be `Invariant`.
+///
```

Comment:
```suggestion
```
I think this comment isn't particularly helpful :thinking: I think having comments explaining the used variance at the callsite instead is better. Alternatively make sure that the comment is talking about "when validating assignments, the variance should be `Covariant`. When checking runtime mir..."